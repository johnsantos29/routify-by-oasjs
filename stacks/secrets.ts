import { Config, StackContext } from "@serverless-stack/resources";

export const SecretsStack = ({ stack }: StackContext) => {
    const VPC_ID = new Config.Secret(stack, "VPC_ID");
    return { vpcId: VPC_ID };
};
