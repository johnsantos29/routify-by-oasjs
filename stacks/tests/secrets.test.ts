import { test, expect } from "vitest";
import { Template } from "aws-cdk-lib/assertions";
import { App, getStack } from "@serverless-stack/resources";
import { SecretsStack } from "../secrets";

test("the creation of SecretsStack", () => {
    const app = new App();
    app.stack(SecretsStack);

    const stack = getStack(SecretsStack);
    const template = Template.fromStack(stack);

    expect(template.toJSON()).toMatchSnapshot();
});
