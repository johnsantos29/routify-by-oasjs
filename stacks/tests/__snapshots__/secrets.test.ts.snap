// Vitest Snapshot v1

exports[`the creation of SecretsStack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomResourceHandlerE8FB56BA": {
      "DependsOn": [
        "CustomResourceHandlerServiceRoleDefaultPolicy73F93F56",
        "CustomResourceHandlerServiceRole41AEC181",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-my-account-us-east-1",
          "S3Key": "ad85de0764d9cc133f45a9f81e4a652509a0bfa17ec6c748e7953e5f125f3172.zip",
        },
        "Handler": "index.handler",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "CustomResourceHandlerServiceRole41AEC181",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
        "Tags": [
          {
            "Key": "sst:app",
            "Value": "my-app",
          },
          {
            "Key": "sst:stage",
            "Value": "dev",
          },
        ],
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomResourceHandlerServiceRole41AEC181": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "sst:app",
            "Value": "my-app",
          },
          {
            "Key": "sst:stage",
            "Value": "dev",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomResourceHandlerServiceRoleDefaultPolicy73F93F56": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutObject",
                "s3:DeleteObject",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:s3:::undefined/*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomResourceHandlerServiceRoleDefaultPolicy73F93F56",
        "Roles": [
          {
            "Ref": "CustomResourceHandlerServiceRole41AEC181",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "StackMetadata": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "App": "my-app",
        "Metadata": [
          {
            "addr": "c82d9896a68cf815f734e61811173ef461a82b9bea",
            "data": {
              "name": "VPC_ID",
            },
            "id": "VPC_ID",
            "stack": "dev-my-app-SecretsStack",
            "type": "Secret",
          },
        ],
        "SSTVersion": "1.15.8",
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomResourceHandlerE8FB56BA",
            "Arn",
          ],
        },
        "Stack": "dev-my-app-SecretsStack",
        "Stage": "dev",
      },
      "Type": "Custom::StackMetadata",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
